2996_reflection/reflection.md

```diff
@@ -3008,6 +3008,63 @@ Change the grammar for `$operator-or-punctuator$` in paragraph 1 of [lex.operato
 ```
 :::
 
+### [basic.pre]{.sref} Preamble {-}
+
+Add type aliases and namespace aliases to the list of entities in paragraph 3. As drive-by fixes, remove "variable", "object", "reference", and "template specialization"; replace "class member" with "non-static data member", since all other cases are subsumed by existing one. Add "template parameters" and "`$init-capture$`s", which collectively subsume "packs".
+
+::: std
+[3]{.pnum} An _entity_ is a [value, object, reference]{.rm} [variable,]{.addu} structured binding, function, enumerator, type, [type alias]{.addu}, [class]{.rm} [non-static data]{.addu} member, bit-field, template, template specialization, namespace, [namespace alias, template parameter]{.addu}, or [`$init-capture$`]{.addu} [pack]{.rm}.
+
+:::
+
+Introduce a notion of an "underlying entity" in paragraph 5, and utilize it for the definition of a name "denoting" an entity. Type aliases are now entities, so also modify accordingly.
+
+::: std
+[5]{.pnum} Every name is introduced by a _declaration_, which is a
+
+* [#.#]{.pnum} `$name-declaration$`, `$block-declaration$`, or `$member-declaration$` ([dcl.pre]{.sref}, [class.mem]{.sref}),
+
+[...]
+
+* [#.14]{.pnum} implicit declaration of an injected-class-name ([class.pre]{.sref}).
+
+[The _underlying entity_ of an entity is that entity unless otherwise specified. A name _denotes_ the underlying entity of the entity declared by each declaration that introduces the name. [An entity `$E$` is denoted by the name (if any) that is introduced by a declaration of `$E$` or by a `$typedef-name$` introduced by a declaration specifying `$E$`.]{.rm}]{.addu}
+
+[[Entities whose underlying entity is not themselves include type aliases and namespace aliases.]{.note}]{.addu}
```

Comment:
I'd phrase this as like "A type alias and a namespace alias are examples of entities whose underlying entity is distinct from itself."? Something like that.